impl/src/main/java/org/ehcache/internal/store/disk/DiskStore.java

```diff
@@ -1005,8 +1007,8 @@ public void remove() {
       TimeSource timeSource = timeSourceConfig != null ? timeSourceConfig.getTimeSource() : SystemTimeSource.INSTANCE;
       
       SerializationProvider serializationProvider = serviceProvider.findService(SerializationProvider.class);
-      Serializer<Element> elementSerializer = serializationProvider.createSerializer(Element.class, storeConfig.getClassLoader());
-      Serializer<Object> objectSerializer = serializationProvider.createSerializer(Object.class, storeConfig.getClassLoader());
+      Serializer<Element> elementSerializer = serializationProvider.createSerializer(Element.class, storeConfig.getClassLoader(), DefaultSerializationProviderConfiguration.find(DefaultSerializationProviderConfiguration.Type.KEY, serviceConfigs));
```

Comment:
Yuck, we call a static method on a specific provider implementation. Why can't the `SerializationProvider` offer this method?